tests/unit/service/V2/test_database.py

```diff
@@ -128,3 +129,50 @@ def test_database_delete_busy_engine(
 
     with raises(AttachedEngineInUseError):
         mock_database.delete()
+
+
+def test_database_update(
+    httpx_mock: HTTPXMock,
+    system_engine_no_db_query_url: str,
+    mock_engine: Engine,
+    mock_database: Database,
+    resource_manager: ResourceManager,
+    query_statistics: Dict[str, Any],
+):
+    def update_query_callback(request, **kwargs):
+        assert "ALTER DATABASE" in request.read().decode()
+        assert "new description" in request.read().decode()
+        query_response = {
+            "meta": [{"name": "one", "type": "int"}],
+            "data": [],
+            "rows": 0,
+            "statistics": query_statistics,
+        }
+        return Response(status_code=codes.OK, json=query_response)
+
+    httpx_mock.add_callback(
+        update_query_callback, url=system_engine_no_db_query_url, method="POST"
+    )
+    # mock get_attached_engines method to return an engine in a running state
+    mock_database.get_attached_engines = lambda: [mock_engine]
+
+    mocked_service = resource_manager.engines
+    mock_database._service = mocked_service
+
+    # Call the update method
```

Comment:
Does this (and similar) comment(s) help to understand the code? IMHO line
```
updated_database = mock_database.update(description="new description")
```
updates description of mock DB and this is obvious even for person that has written 2 python lines in his like (like me) ;)

Reply:
You're right. Had those left over from the design phase.